// gitbook-plugin-tesela/index.js
var header = "<div class='top-bar'>  \
       <a href='/'><img src='https://www.teselagen.com/img/teselagen_toplogo.png'></a></li>  \
</div>";


var footer = "<div id='footer'> \
      <div id='footer_links' class='large-12 small-12 small-centered large-centered'> \
        <ul class='footer-ul'>                         \
          <li><a href='/about'>About</a></li>          \
          <li><a href='/careers'>Careers</a></li>      \
          <li><a href='/faq'>FAQ</a></li>              \
          <li><a href='/terms'>Terms</a></li>          \
          <li><a href='/privacy'>Privacy</a></li>      \
          <li><a href='/news'>News &amp; Press</a></li> \
          <li><a href='http://help.teselagen.com' target='_blank'>Documentation</a></li>  \
          <li><a href='/contact'>Contact</a></li>        \
        </ul> \
        <ul class='footer-ul social-footer'> \
          <li><a href='https://www.facebook.com/Teselagen' target='_blank' class='social-footer-link'><img src='https://www.teselagen.com/img/icons/img-footer-facebook.png'></a></li> \
          <li><a href='https://twitter.com/teselagen' target='_blank' class='social-footer-link'><img src='https://www.teselagen.com/img/icons/img-footer-twitter.png'></a></li> \
          <li><a href='http://www.linkedin.com/company/2838822' target='_blank' class='social-footer-link'><img src='https://www.teselagen.com/img/icons/img-footer-linkedin.png'></a></li> \
          <li><a href='https://www.youtube.com/channel/UC9gW_S5i13WT68BiB3hQAhA' target='_blank' class='social-footer-link'><img src='https://www.teselagen.com/img/icons/img-footer-youtube.png'></a></li> \
        </ul> \
      </div> \
    </div>";

var footer2 = '<script>document.write("<scr"+"ipt> $( document ).ready(function() { $(\'.summary > .chapter > a\').click(function() {   $(this).parent().find(\'> ul\').slideToggle();return false;});  $(\'.chapter \').find(\'> a\').first().on(\'click\', function(){  window.location = \'/manual/\';});   $(\'.summary > .chapter > a\').parent().find(\'> ul\').hide();  }); </scr" + "ipt>") </script>';

var footer3 = '<script>window.addEventListener("DOMContentLoaded", function() { $(\'.summary > .chapter > a\').click(function() {   $(this).parent().find(\'> ul\').slideToggle(); return false });  $(\'.chapter \').find(\'> a\').first().on(\'click\', function(){  window.location = \'/manual/\';});   $(\'.summary > .chapter > a\').parent().find(\'> ul\').hide();}, false); </script>';



module.exports = {
    book: {
        assets: "./book",
        //js: [
	//     "tesela.js"
        //],
        css: [
            "tesela.css"
	],
	html: {
        //     "html:start": function() {
        //         return header;
        //     },
             "html:end": function() {
                 return footer3;
             }
	}

    },
    hooks: {
        init: function() {
            console.log("init tesela plugin/theme!");
        },
        finish: function() {
            console.log("finish plugin/theme!");
        }
    }
};
